datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -1216,11 +1216,7 @@ fn eq_dyn_null(
     // implementation
     // <https://github.com/apache/datafusion/issues/10749>
     if left.data_type().is_nested() && null_equals_null {
-        let cmp = make_comparator(left, right, SortOptions::default())?;
-        let len = left.len().min(right.len());
-        let values = (0..len).map(|i| cmp(i, i).is_eq()).collect();
-        let nulls = NullBuffer::union(left.nulls(), right.nulls());
-        return Ok(BooleanArray::new(values, nulls));
+        return Ok(compare_op_for_nested(&Operator::Eq, &left, &right)?);
```

Comment:
the point of the PR is to use the pre-existing function that already handles this

Reply:
I think we can use `apply_cmp_for_nested` and keep `compare_op_for_nested` private here.